### PR TITLE
Top Secret support for user profile extraction.

### PR DESCRIPTION
**Pull request by jordanfilteau1995**
This pull request adds Top Secret functions that will help in the implementation of a security API.

Support for user profile extraction is also present.

Signed-off-by: Jordan Filteau <jfilteau@rocketsoftware.com>

### CODE REVIEW

File: c/zis/services/secmgmttss.c [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+/* A helper function which finds the line where the value where we are looking for starts
+ * and returns a pointer to it. This is useful for when the caller wants to start from
+ * a particular value because they either reached the maximum they could return in a 
+ * request, or if they want to information on a single user. It also returns pointers
+ * to the lastByteOffset and the blockBytesExtracted so when continuing to parse,
+ * we still have the right place through the output.
+ */
+static const RadminVLText *findLineOfStart(const RadminVLText *currentLine, const char *searchValue, 
+                                           int searchValueMaxLength, const char *searchKey,
+                                           const unsigned int *lastByteOffset, unsigned int *blockBytesExtracted,
```
Comment (ifakhrutdinov) [inline c/zis/services/secmgmttss.c]: Why don't you just pass the value instead of the address of ```lastByteOffset```?

Comment (ifakhrutdinov) [inline c/zis/services/secmgmttss.c]: A better name for ```blockBytesExtracted``` would be ```offsetInBlock``` and for ```lastByteOffset``` is ```maxOffsetInBlock```, that will indicate the purpose better.

File: c/zis/services/tssparsing.c [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+#include <metal/stdio.h>
+#include <metal/string.h>
+
+#ifndef METTLE
+#error Non-metal C code is not supported
+#endif
+
+#include "zowetypes.h"
+#include "alloc.h"
+#include "crossmemory.h"
```
Comment (ifakhrutdinov) [inline c/zis/services/tssparsing.c]: Why do you need ```crsossmemory``` here?

File: h/zis/services/tssparsing.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+ */
+typedef struct KeyData_t {
+  unsigned int startPos;
+  unsigned int endPos;
+  unsigned int delimPos;
+  char keyName[MAX_KEY_LENGTH];
+  int keyLength;
+} KeyData;
+
+bool isErrorDetected(const char *str, int len);
```
Comment (ifakhrutdinov) [inline h/zis/services/tssparsing.h]: Can you please prefix these so that there is no name collision? E.g. ```tssutil```

**Comment by jordanfilteau1995** [comment]
@ifakhrutdinov Changes have been made. Please review at your convenience. Thank you.

**Comment by jordanfilteau1995** [comment]
> 
> 
> My other concerns:
> 
>     * The performance will be potentially bad when we use `startUserID` because we have to reach it manually
> 
>     * What if the start user gets removed or not found? It seems we won't extract anything

Yeah, there isn't a callable service, so I have to find it manually. There isn't really any counter play for that which I can think of.

File: c/zis/services/secmgmtUtils.c [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+#ifndef METTLE
+#error Non-metal C code is not supported
+#endif
+
+#include "zowetypes.h"
+#include "alloc.h"
+#include "crossmemory.h"
+#include "radmin.h"
+#include "recovery.h"
```
Comment (ifakhrutdinov) [inline c/zis/services/secmgmtUtils.c]: What do you need ```recovery.h``` for here?

File: c/zis/services/secmgmtUtils.c [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+#include "zos.h"
+
+bool getCallerUserID(RadminUserID *caller) {
```
Comment (ifakhrutdinov) [inline c/zis/services/secmgmtUtils.c]: Please use a prefix. For example, ```secmgmtGetCallerUserID``` or ```secutilGetCallerUserID```

File: c/zis/services/secmgmtUtils.c [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
```
Comment (ifakhrutdinov) [inline c/zis/services/secmgmtUtils.c]: Put a newline here.

**Pull request closed without merging**